code/modules/tram/tram.dm

```diff
@@ -229,8 +247,38 @@
 		if(A.light_range)
 			A.set_light()
 	gen_collision() //Generate collision again
+	for(var/atom/movable/A in crash_list)
+		if(crash_list[A] == dir)
+			handle_crash(A,dir)
+		else
+			crash_list -= A
 	return 1
 
+
+obj/tram/tram_controller/proc/handle_crash(var/atom/movable/A,var/dir)
+	if(!istype(A)) return
+	if((A in gibbed)) return //so we don't gib things multiple times
+	if(istype(A,/mob/living))
+		if(istype(A,/mob/living/carbon))
+			if(A:lying)
+				crash_list -= A
+				A:gib()
+				gibbed += A
+			else //if it's not lying down, push it and break its face
+				A:apply_damage(5) //brute by default, 50 damage in one second
+				if(prob(2)) A:Weaken(2) //RNGesus is not in your favor, RIP buddy...
+				step_towards(A, get_step(get_turf(A),dir))
+				crash_list -= A
+		else
+			crash_list -= A
+			A:gib() //gib all non-carbon things like simple mobs
```

Comment:
Potentially, yes. Maybe make it `/atom/movable/proc/handle_tram_crash(obj/tram/tram_controller/tram, dir)` and just call `A.handle_tram_crash(src, dir)`